### DISCUSSION TITLE
questions on architecture-design.md of 1.3.9

### DISCUSSION THREAD

**Discussion by RedemptionC** [apache/dolphinscheduler#7835] [General]
[this doc says](https://github.com/apache/dolphinscheduler-website/blob/master/docs/zh-cn/1.3.9/user_doc/architecture-design.md#23-%E6%9E%B6%E6%9E%84%E8%AE%BE%E8%AE%A1%E6%80%9D%E6%83%B3) that task queue is based on ZK
```
Task Queue

提供任务队列的操作，目前队列也是基于Zookeeper来实现。由于队列中存的信息较少，不必担心队列里数据过多的情况，实际上我们压测过百万级数据存队列，对系统稳定性和性能没影响。
```
while it seems that it's not based on ZK in 1.3.x,as I consulted in the wechat group
![image](https://user-images.githubusercontent.com/17563754/148251227-f19f84c6-79b3-4ada-aae0-15cbfb61a6f0.png)

what's the truth? Is it that the doc should be updated? 
I find it a little hard to figure out the architecture design by just studying the code

Any advice and information will be appreciated

**Comment by Lucaszlei** [comment]
The truth is as dailidong said, zk is only used for the registration of user master and worker in version 1.3.x

> what's the truth? Is it that the doc should be updated?

I think the description in the document is inappropriate

**Reply by zhongjiajie** [reply]
Thanks @SunnyZ-L for clarification, and both @RedemptionC @SunnyZ-L could submit PR to correct this if you like to

**Reply by RedemptionC** [reply]
@zhongjiajie I'd like to,but it takes time for me to figure it out how it works,as the doc sometimes doesn't provide valid information
For example,it seems that in 1.3.9 arch.md,it says:
```
WorkerServer

WorkerServer也采用分布式无中心设计理念，WorkerServer主要负责任务的执行和提供日志服务。 WorkerServer服务启动时向Zookeeper注册临时节点，并维持心跳。 WorkerServer基于netty提供监听服务。

该服务包含：

FetchTaskThread主要负责不断从Task Queue中领取任务，并根据不同任务类型调用TaskScheduleThread对应执行器。

LoggerServer是一个RPC服务，提供日志分片查看、刷新和下载等功能
```
while there isn't TaskScheduleThread in 1.3.9 release

**Reply by zhongjiajie** [reply]
@RedemptionC Thanks for your pay attention to DolphinScheduler. TBO, I also do not know some inside designs(maybe most of the inside designs). But you could still ping me if have some problems. I would like to help you and ping others who know the inside designs
